src/datajudge/db_access.py

```diff
@@ -294,9 +294,12 @@ class DataReference:
     def __init__(
         self,
         data_source: DataSource,
-        columns: list[str] = None,
-        condition: Condition = None,
+        columns: Optional[list[str]] = None,
+        condition: Optional[Condition] = None,
     ):
+        if columns is not None and not isinstance(columns, list):
```

Comment:
I think we should generalize a bit the `isinstance` call depending on the answer in [here](https://github.com/Quantco/datajudge/pull/155/files#r1223975407),

If the answer is yes, then here probably we should compare to something like `Sequence`. If the answer is no, then we should compare to `Collection`.

I do realize that there will be a slight mismatch between the annotation in the function signature and what's happening inside at runtime, but we could fix the annotation in a second step.

Reply:
Fair point. Then let's tackle this in a separate discussion.